src/components/views/right_panel/UserInfo.tsx

```diff
@@ -1024,7 +1024,7 @@ const PowerLevelEditor: React.FC<{
 
         const myUserId = cli.getUserId();
         const myPower = powerLevelEvent.getContent().users[myUserId];
-        if (myPower && parseInt(myPower) === powerLevel) {
+        if (myPower && parseInt(myPower) === powerLevel && myUserId !== target) {
```

Comment:
Similarly to my other comment, this should check whether the power level actually increased (otherwise it's wrong to say that you're "promoting the user")

Reply:
Thanks @robintown, I changed the logic and it works. Thanks for the help.